client/src/main/java/io/cloudsoft/winrm4j/client/ShellCommand.java

```diff
@@ -46,14 +47,21 @@
     private SelectorSetType shellSelector;
 
     private String operationTimeout;
+    /**
+     * Define if a new Receive request will be send when the server returns a fault with the code
+     * {@link #WSMAN_FAULT_CODE_OPERATION_TIMEOUT_EXPIRED}.
+     */
+    private Predicate<String> retryReceiveAfterOperationTimeout;
     private final Locale locale;
 
     private int numberOfReceiveCalls;
 
-    public ShellCommand(WinRm winrm, String shellId, String operationTimeout, Locale locale) {
+    public ShellCommand(WinRm winrm, String shellId, String operationTimeout, Predicate<String> retryReceiveAfterOperationTimeout,
```

Comment:
Minor observation: in case anyone is trating `ShellCommand` constructor as part of the public api (?!), I'd be tempted to overload it so there is a constructor with the original signature as well.

But on the other hand, that would be making the code more complicated - defaults being set here and in `WinRmClientBuilder ` constructor, and in `assertFaultCode(SOAPFaultException soapFault, String code)`!